torch_geometric/data/data.py

```diff
@@ -842,6 +842,9 @@ def _put_edge_index(self, edge_index: EdgeTensorType,
         attr_val = edge_tensor_type_to_adj_type(edge_attr, edge_index)
         setattr(self, attr_name, attr_val)
 
+        # Set edge attributes:
+        setattr(self, f'{attr_name}_edge_attr', edge_attr)
```

Comment:
How does this work when we add multiple `edge_attr` to a single object? Won't we need to condition that on layout?
```python
self._edge_attrs = {
    EdgeLayout.COO: copy.copy(edge_attr)
}
```
I would also make this a private attribute to not expose it as part of `print(data)`.

Reply:
`attr_name` should indicate what the layout is (either `edge_index`, `adj`, or `adj_t`), right?

Reply:
You are right, missed that. Shall we use `edge_attr.layout.value` for this? + Also make it private.

Reply:
Sure, sounds good